PSKoans/Koans/IntroTraining/AboutBinary.koans.ps1

```diff
@@ -0,0 +1,180 @@
+using module PSKoans
+[Koan(Position = 109)]
+param()
+<#
+    Binary
+
+    Binary is a base 2 number system, and only uses two numbers: 0 and 1.
+
+    Binary is the basis of all computing and computational storage. It's simple & straightforward,
+    and directly represents the underlying hardware states, which can only be on (1) or off (0).
+
+    It's useful to know binary when working with PowerShell to understand how integers are represented,
+    which is our next Koan. It's also extremely useful knowledge if you plan on delving deeper into computing;
+    the knowledge is applicable to networking, programming, data science, databases, and essentially every
+    other area of computing.
+
+    What is represents is a base 2 number. 
+
+    For example, the below binary represents a "byte" of data, 
+    which contains 8 "bits". Each bit being either 1 or 0 as mentioned earlier.
+
+    So:
+
+     2^7 | 2^6 | 2^5 | 2^4 | 2^3 | 2^2 | 2^1 | 2^0
+     128 |  64 |  32 |  16 |   8 |   4 |   2 |   1 
+    -----+-----+-----+-----+-----+-----+-----+-----
+       0     0     0     0     0     0     0     1 
+    
+    Or:
+
+    00000001
+
+    Is actually 1
+
+    And 00000011
+
+    Is three. As both 1 and 2 are on, and 1 + 2 is equal to 3.
+
+#>
+
+Describe "Binary conversion" {
+
+    It "Bit conversion" {
+    <#
+        When converting from bit to a boolean try and guess what each would be
+    #>
+
+    # What would 0 be if converted to be boolean?
+    # Replace __ with either $true or $false
+    $____ -as [int] | Should -Be 0
+
+    # What would 1 be if converted to be boolean?
+    # Replace __ with either $true or $false
+    $__ -as [int] | should -Be 1
+
+    }
+
+    It "Nibble to Int conversion" {
+        <#
+            Convert the following  nibbles into full numbers
+        #>
+        
+        # Replace __ with the decimal value of 1111
+        # E.G. __ becomes 1234
+        $Binary = "1111"
+        $Value = __
+        $Value | Should -Be ([Convert]::ToInt32($Binary, 2))
+        
+        # Replace __ with the decimal value of 1000
+        # E.G. __ becomes 1234
+        $Binary = "1000"
+        $Value = __
+        $Value | Should -Be ([Convert]::ToInt32($Binary, 2))
+
+        # Replace __ with the decimal value of 0010
+        # E.G. __ becomes 1234
+        $Binary = "0010"
+        $Value = __
+        $Value | Should -Be ([Convert]::ToInt32($Binary, 2))
+
+        # Replace __ with the decimal value of 1001
+        # E.G. __ becomes 1234
+        $Binary = "1001"
+        $Value = __
+        $Value | Should -Be ([Convert]::ToInt32($Binary, 2))
+
+    }
+
+    It "Int to Nibble conversion" {
+        <#
+            Convert the following  full numbers into nibbles
```

Comment:
```suggestion
            Convert the following integers into their binary representation.
```

I think this section can be merged with the next one, since there's not an actual data type that is a nibble; the smallest integer data type (ignoring booleans) is a byte. So while a nibble is a thing in the number theory, I think it would suffice to have these as the starting section of the next `It` block, and just group them as `It 'Binary to integer conversions'` or something along those lines. 🙂

Reply:
We could merge these with the next ```It``` block, but this is named ```Byte to Int``` and the examples given are 4 bits. It might give the impression that a byte is 4-bits and potentially confuse learners about how many bits they're expected to use in the answer.

Instead why not rename the blocks and change the wording slightly?

Lines 57 to 60 will become:

```PowerShell    
    It "4-bits to integer conversion" {
        <#
            Convert the following 4-bits into their decimal equivalent.
        #>
```
Lines 85 to 88 will become:

```PowerShell
    It "4-bits to integer conversion" {
        <#
            Convert the following 4-bits numbers into nibbles
        #>
```

Thoughts?

Reply:
You could just name it `Binary to integer conversions` and call it a day 😉 

The reason I'd say it's probably better to keep them together is that if we separate them, we create an impression that there is a kind of number that only uses 4 bits, and there isn't, really. The lowest .NET works with is bytes, which are 8. 🤷‍♂

Reply:
Renamed and calling it a day 👅 Also merged into just three ```It``` blocks.